test/PublicApiTests/BaseLine/Microsoft.OData.PublicApi.net45.bsl

```diff
@@ -7498,74 +7470,6 @@ public sealed class Microsoft.OData.UriParser.Aggregation.GroupByTransformationN
     Microsoft.OData.UriParser.CollectionNode Source  { public get; }
 }
 
-public enum Microsoft.OData.UriParser.Validation.Severity : int {
-    Error = 3
-    Info = 1
-    Undefined = 0
-    Warning = 2
-}
-
-public abstract class Microsoft.OData.UriParser.Validation.ODataUrlValidationRule {
-    protected ODataUrlValidationRule ()
-
-    bool IncludeImpliedProperties  { public get; public set; }
-    string RuleName  { public get; protected set; }
-
-    internal abstract System.Type GetRuleType ()
-    internal abstract void Validate (Microsoft.OData.UriParser.Validation.ODataUrlValidationContext context, object component)
-}
-
-[
-ExtensionAttribute(),
-]
-public sealed class Microsoft.OData.UriParser.Validation.ODataUrlValidationExtensions {
-    [
-    ExtensionAttribute(),
-    ]
-    public static bool ValidateODataUrl (System.Uri uri, Microsoft.OData.Edm.IEdmModel model, Microsoft.OData.UriParser.Validation.ODataUrlValidationRuleSet rules, out System.Collections.Generic.IEnumerable`1[[Microsoft.OData.UriParser.Validation.ODataUrlValidationMessage]]& validationMessages)
-}
-
-public sealed class Microsoft.OData.UriParser.Validation.ODataUrlValidationMessageCodes {

```

Comment:
>ODataUrlValidationMessageCodes [](start = 57, length = 30)

This is an enum for EdmError and strings for ValidationMessage.

Reply:
Can you share detail? I assume that we should change it as Enum type ? right

---
In reply to: [445653762](https://github.com/OData/odata.net/pull/1826#discussion_r445653762) [](ancestors = 445653762)

Reply:
This was a note to myself. For the Edm Validator, EdmErrorCode is an enum. Here it is a string.  I actually think that's okay, as it allows custom rules to return new error codes without us having to update the enum, I just wanted to call it out.

---
In reply to: [445691615](https://github.com/OData/odata.net/pull/1826#discussion_r445691615) [](ancestors = 445691615,445653762)